test/dotnet-publish.Tests/Microsoft.DotNet.Tools.Publish.Tests.cs

```diff
@@ -74,40 +97,25 @@ public void PublishOptionsTest(string framework, string runtime, string config,
         [Fact]
         public void ProjectWithContentsTest()
         {
-            // create unique directories in the 'temp' folder
-            var testDir = Temp.CreateDirectory();
-
-            var testAppDir = Path.Combine(_testProjectsRoot, "TestAppWithContents");
+            TestScenario scenario = TestScenario.Create(Path.Combine(_testProjectsRoot, "TestAppWithContents"), skipRestore: true);
```

Comment:
This line is pretty verbose:

```
TestScenario scenario = TestScenario.Create(Path.Combine(_testProjectsRoot, "TestAppWithContents"), skipRestore: true);
```

Maybe a builder pattern would be better? Also more defaults. Why are we passing in the `Path.Combine`?